File: app/src/main/java/com/example/myapplication/inner/MainActivity.java
package com.example.myapplication.inner;

import android.os.AsyncTask;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.example.myapplication.R;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;

import okhttp3.Call;
import okhttp3.Callback;
import okhttp3.MediaType;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.RequestBody;
import okhttp3.Response;


public class MainActivity extends AppCompatActivity implements View.OnClickListener {


    TextView txtString;

    public String url = "https://api.staging.okcredit.in/v1.0/version?lang=en";
    public String POST_URL_AB = "https://ab.staging.okcredit.in/v1/GetProfile";
    public String AB_BODY = "{\n" +
            "    \"device_id\": \"86c20472-5ff6-4d23-b1f6-546bb041b7d3\",\n" +
            "    \"merchant_id\": \"\"\n" +
            "}";

    public static final MediaType JSON = MediaType.parse("application/json; charset=utf-8");

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.postOkhttp).setOnClickListener(this);
        findViewById(R.id.postRetrofit).setOnClickListener(this);

        txtString = findViewById(R.id.txtString);
    }

    void postRequestOkHttp(String postUrl, String postBody) throws IOException {

        OkHttpClient client = new OkHttpClient();

        RequestBody body = RequestBody.create(JSON, postBody);

        Request request = new Request.Builder()
                .url(postUrl)
                .post(body)
                .build();

        client.newCall(request).enqueue(new Callback() {
            @Override
            public void onFailure(Call call, IOException e) {
                call.cancel();
            }

            @Override
            public void onResponse(Call call, Response response) throws IOException {

                runOnUiThread(() -> {
                    try {
                        txtString.setText(response.body().string());
                    } catch (IOException e) {
                        e.printStackTrace();
                    }
                });
            }
        });
    }


    void run() throws IOException {

        OkHttpClient client = new OkHttpClient();

        Request request = new Request.Builder()
                .url(url)
                .build();

        client.newCall(request).enqueue(new Callback() {
            @Override
            public void onFailure(Call call, IOException e) {
                call.cancel();
            }

            @Override
            public void onResponse(Call call, Response response) throws IOException {

                final String myResponse = response.body().string();

                MainActivity.this.runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        try {

                            JSONObject json = new JSONObject(myResponse);
                            txtString.setText(json.toString());
                        } catch (JSONException e) {
                            e.printStackTrace();
                        }
                    }
                });

            }
        });
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.postOkhttp:
                try {
                    postRequestOkHttp(POST_URL_AB, AB_BODY);
                } catch (IOException e) {
                    e.printStackTrace();
                }
                break;
            case R.id.postRetrofit:
                break;

        }
    }

    public class OkHttpHandler extends AsyncTask<String, Void, String> {

        OkHttpClient client = new OkHttpClient();

        @Override
        protected String doInBackground(String... params) {

            Request.Builder builder = new Request.Builder();
            builder.url(params[0]);
            Request request = builder.build();

            try {
                Response response = client.newCall(request).execute();
                return response.body().string();
            } catch (Exception e) {
                e.printStackTrace();
            }
            return null;
        }

        @Override
        protected void onPostExecute(String s) {
            super.onPostExecute(s);
            txtString.setText(s);
        }
    }


}

File: app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import okhttp3.*
import okhttp3.MediaType.Companion.toMediaType
import java.io.IOException
import java.util.*

class MainActivity : AppCompatActivity(), View.OnClickListener {
    var txtString: TextView? = null

    companion object {
        val JSON = "application/json; charset=utf-8".toMediaType()

        private var POST_URL_AB = "https://ab.staging.okcredit.in/v1/GetProfile"
        private var AB_BODY = """{
                "device_id": "86c20472-5ff6-4d23-b1f6-546bb041b7d3",
                "merchant_id": ""
        }"""
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViewById<View>(R.id.postOkhttp).setOnClickListener(this)
        findViewById<View>(R.id.postRetrofit).setOnClickListener(this)
        findViewById<View>(R.id.postAsyncTask).setOnClickListener(this)
        txtString = findViewById(R.id.txtString)
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.postOkhttp -> try {
                postRequestOkHttp(POST_URL_AB, AB_BODY)
            } catch (e: IOException) {
                e.printStackTrace()
            }
            R.id.postRetrofit -> {
                postRequestRetrofit()
            }
            R.id.postAsyncTask -> {
                postAsyncTask()
            }
        }
    }

    private fun postAsyncTask() {
        val postRequest = AsynTaskPost.POSTRequest()

        postRequest.execute(POST_URL_AB, AB_BODY)
    }

    @Throws(IOException::class)
    fun postRequestOkHttp(postUrl: String, postBody: String) {
        val client = OkHttpClient()
        val body = RequestBody.create(JSON, postBody)
        val request: Request = Request.Builder()
            .url(postUrl)
            .post(body)
            .build()
        client.newCall(request).enqueue(object : Callback {
            override fun onFailure(call: Call, e: IOException) {
                call.cancel()
            }

            @Throws(IOException::class)
            override fun onResponse(call: Call, response: Response) {
                runOnUiThread {
                    try {
                        txtString!!.text = response.body!!.string()
                    } catch (e: IOException) {
                        e.printStackTrace()
                    }
                }
            }
        })
    }


    fun postRequestRetrofit(){
        val apiInterface = ApiInterface.create().getAB(GetProfileRequest(merchant_id = "", device_id = UUID.randomUUID().toString()))

        //apiInterface.enqueue( Callback<List<Movie>>())
        apiInterface.enqueue(object:retrofit2.Callback<GetProfileResponse> {
            override fun onResponse(
                call: retrofit2.Call<GetProfileResponse>,
                response: retrofit2.Response<GetProfileResponse>
            ) {
                try {
                    txtString!!.text = response.body().toString()
                } catch (e: IOException) {
                    e.printStackTrace()
                }
            }

            override fun onFailure(call: retrofit2.Call<GetProfileResponse>, t: Throwable) {
            }

        })
    }
}